packages/walletconnect/src/index.spec.ts

```diff
@@ -21,23 +21,25 @@ const chains = [1, 2, 3]
 
 describe('WalletConnect', () => {
   const wc2RequestMock = jest.fn()
+  let wc2InitMock: jest.Mock
 
   beforeAll(() => {
     const wc2EnableMock = jest.fn().mockResolvedValue(accounts)
-
     // @ts-ignore
     // TypeScript error is expected here. We're mocking a factory `init` method
     // to only define a subset of `EthereumProvider` that we use internally
-    jest.spyOn(EthereumProvider, 'init').mockImplementation(async (opts) => ({
+    wc2InitMock = jest.spyOn(EthereumProvider, 'init').mockImplementation(async (opts) => ({
       // we read this in `enable` to get current chain 
       accounts,
       chainId: opts.chains[0],
-      // non-null `session` indicates we have a connected wallet 
-      session: wc2EnableMock.mock.calls.length,
+      // session is an object when connected, undefined otherwise
+      get session() {
+        return wc2EnableMock.mock.calls.length > 0 ? {} : undefined
+      },
       // methods used in `activate` and `isomorphicInitialize`
       enable: wc2EnableMock,
       request: wc2RequestMock,
-      // basic EventEmmiter
+      // basic EventEmitter
       on() {},
       off() {},
```

Comment:
Does `off` need to be implemented? I thought just `removeListener`.

Reply:
Doesn't have to be "per spec". However, WalletConnect provider defines some additional methods (such as `off`) and we're relying on `off` in the context of our class (where we know it's WC provider, not EIP 1193)

Reply:
With the future work around writing an EIP-1193 compliant connector, I think we can migrate to `removeListener` to not rely on extraneous properties.